src/util/config.cpp

```diff
@@ -7,86 +7,95 @@ Author: Daniel Kroening, kroening@kroening.com
 \*******************************************************************/
 
 #include <algorithm>
+#include <regex>
 
 #include <util/config.h>
 
 configt config;
 
-void configt::ansi_ct::set_16()
+void configt::ansi_ct::set_data_model(enum data_model dm)
 {
-  bool_width = 1 * 8;
-  int_width = 2 * 8;
-  long_int_width = 4 * 8;
-  char_width = 1 * 8;
-  short_int_width = 2 * 8;
-  long_long_int_width = 8 * 8;
-  pointer_width = 4 * 8;
-  pointer_diff_width = 4 * 8;
-  single_width = 4 * 8;
-  double_width = 8 * 8;
-  long_double_width = 8 * 8;
+  uint64_t m = dm;
+
+  char_width = m & 0xff, m >>= 8;
+  short_int_width = m & 0xff, m >>= 8;
+  int_width = m & 0xff, m >>= 8;
+  long_int_width = m & 0xff, m >>= 8;
+  pointer_width = m & 0xff, m >>= 8;
+  word_size = m & 0xff, m >>= 8;
+  long_double_width = m & 0xff, m >>= 8;
+
```

Comment:
Can we avoid comma operators here? You can use a lambda to perform the two operations.

Reply:
I've used comma for readability. clang-format otherwise makes a mess out of this function. What is the reason for avoiding the comma operator in general?

Reply:
Code-style reasons mainly.

Reply:
I have doubts that this twice as long code improves readability:
```diff
-  uint64_t m = dm;
-
-  char_width = m & 0xff, m >>= 8;
-  short_int_width = m & 0xff, m >>= 8;
-  int_width = m & 0xff, m >>= 8;
-  long_int_width = m & 0xff, m >>= 8;
-  pointer_width = m & 0xff, m >>= 8;
-  word_size = m & 0xff, m >>= 8;
-  long_double_width = m & 0xff, m >>= 8;
+  struct
+  {
+    uint64_t m;
+    unsigned operator()()
+    {
+      unsigned r = m & 0xff;
+      m >>= 8;
+      return r;
+    };
+  } next = {dm};
+
+  char_width = next();
+  short_int_width = next();
+  int_width = next();
+  long_int_width = next();
+  pointer_width = next();
+  word_size = next();
+  long_double_width = next();
```
Having control over whitespace and aligning all the `=` symbols would help a lot more; but is not allowed by the current clang-format settings.